File: leet11/leet11.py
#https://leetcode.com/problems/container-with-most-water/submissions/

from sys import stdin
from typing import List
input = stdin.readline

class Solution:
    def maxArea(self, height: List[int]) -> int:
        left: int = 0
        right: int = len(height) - 1
        all_areas = []
        while left < right:
            if height[left] < height[right]:
                all_areas.append((right - left) * height[left])
                left += 1
            else:
                all_areas.append((right - left) * height[right])
                right -= 1
        return max(all_areas)

File: leet11/leet11.cpp
// https://leetcode.com/problems/container-with-most-water/
#include <vector>
using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0, right = height.size() - 1, max_area = 0, cur_area;
        while(left < right){
            cur_area = min(height[left], height[right]) * (right - left);
            max_area = max(cur_area, max_area);
            if(height[left] < height[right]){
                left++;
            }
            else{
                right--;
            }
        }
        return max_area;
    }
};